pandas/core/indexes/datetimelike.py

```diff
@@ -837,13 +844,6 @@ def _union(self, other, sort):
     # Join Methods
     _join_precedence = 10
 
-    _inner_indexer = _join_i8_wrapper(libjoin.inner_join_indexer)
```

Comment:
I think remove L836 & 837 and move L838 to same location makes sense

Reply:
Tried that, moving 838 breaks things bc it changes PeriodIndex._join_precedence